test/e2e/cluster_lifecycle_test.go

```diff
@@ -0,0 +1,261 @@
+			}).GetARMResourcesClientFactoryOrDie(ctx)
+			resourceGroupClient = armResourcesClientFactory.NewResourceGroupsClient()
+			err = framework.DeleteResourceGroup(ctx, resourceGroupClient, resourceGroup, 60*time.Minute)
+			Expect(err).NotTo(HaveOccurred(), "failed to delete resource group during cleanup")
+			By("Verifying the resource group is deleted")
+			_, err = resourceGroupClient.Get(ctx, resourceGroup, nil)
+			Expect(err).ToNot(BeNil())
+			errMessage := fmt.Sprintf("The resource group '%s' could not be found.", resourceGroup)
+			Expect(err.Error()).To(ContainSubstring(errMessage))
+		}()
+
+		By("Create a new resource group for the cluster")
+		// Use the framework's test context for resource group operations since they don't need the systemData policy
+		testCtx = framework.NewInvocationContext()
+		armResourcesClientFactory = testCtx.(interface {
+			GetARMResourcesClientFactoryOrDie(context.Context) *armresources.ClientFactory
+		}).GetARMResourcesClientFactoryOrDie(ctx)
+		resourceGroupClient = armResourcesClientFactory.NewResourceGroupsClient()
+		resourceGroup = fmt.Sprintf("e2e-lifecycle-%s-rg", uuid.NewString()[:4])
+		_, err := framework.CreateResourceGroup(ctx, resourceGroupClient, resourceGroup, testCtx.(interface{ Location() string }).Location(), 10*time.Minute)
+		Expect(err).NotTo(HaveOccurred(), "failed to create resource group")
+
+		By("Deploying the infrastructure only bicep template")
+		deploymentName := fmt.Sprintf("e2e-lifecycle-%s-deployment", uuid.NewString()[:4])
+
+		// Read the bicep template file from the correct location
+		bicepTemplatePath := "../../e2e-setup/bicep/infra-only.bicep"
+		bicepContent, err := os.ReadFile(bicepTemplatePath)
+		Expect(err).NotTo(HaveOccurred(), "failed to read bicep template")
+
+		// Create deployment parameters for infra-only.bicep
+		deploymentParameters := map[string]string{
+			"clusterName":     clusterName,
+			"persistTagValue": "false",
+		}
+
+		// Use the framework's CreateBicepTemplateAndWait function for consistency
+		deploymentResult, err := framework.CreateBicepTemplateAndWait(
+			ctx,
+			armResourcesClientFactory.NewDeploymentsClient(),
+			resourceGroup,
+			deploymentName,
+			bicepContent,
+			deploymentParameters,
+			45*time.Minute,
+		)
+		Expect(err).NotTo(HaveOccurred(), "failed to complete bicep deployment")
+
+		// Get deployment outputs to retrieve managed identity information
+		deploymentOutputs := deploymentResult.Properties.Outputs
+		Expect(deploymentOutputs).NotTo(BeNil(), "deployment outputs should not be nil")
+
+		// Get deployment outputs to retrieve managed identity information
+		outputs := deploymentOutputs.(map[string]interface{})
+
+		// Extract identityValue and userAssignedIdentitiesValue from outputs
+		identityValue := outputs["identityValue"].(map[string]interface{})
+		userAssignedIdentitiesValue := outputs["userAssignedIdentitiesValue"].(map[string]interface{})
```

Comment:
try using this function here: https://github.com/Azure/ARO-HCP/blob/f68fa8de08934df1e05b4e18dce16a90ea27216a/test/util/framework/deployment_helper.go#L46-L49